pkg/types/validation/installconfig.go

```diff
@@ -824,6 +827,16 @@ func validateCompute(platform *types.Platform, control *types.MachinePool, pools
 			allErrs = append(allErrs, field.Invalid(poolFldPath.Child("fencing"), p.Fencing, "fencing is only valid for control plane"))
 		}
 	}
+
+	if control != nil {
+		controlReplicas = ptr.Deref(control.Replicas, 3)
+	}
+
+	// Validate that exactly 1 worker node is allowed only when a single control plane node is provisioned on platforms None or External
+	if computeReplicas == 1 && controlReplicas != 1 && !(platform.None != nil || platform.External != nil) {
```

Comment:
```suggestion
    // Validate that exactly 1 worker node is allowed only in either the following scenarios:
	// 1. A single control plane node (SNO) is provisioned.
	// 2. Platforms are None or External.
	if computeReplicas == 1 {
		isSNO := controlReplicas == 1
		isAllowedPlatform := platform.None != nil || platform.External != nil
		if !isSNO && !isAllowedPlatform {
```

nit: It seems a bit more verbose, but improves readability? I also edit the comment a bit to match the condition. Previously, it hinted that only SNO + platform none/external is allowed. WTDY?

Reply:
```suggestion
    // Validate that exactly 1 worker node is allowed only in either the following scenarios:
	// 1. A single control plane node (SNO) is provisioned.
	// 2. Platforms are None or External.
	if computeReplicas == 1 {
		isSNO := controlReplicas == 1
		isAllowedPlatform := platform.None != nil || platform.External != nil
		if !isSNO && !isAllowedPlatform {
```

nit: It seems a bit more verbose, but improves readability? WDYT?

Reply:
1+1 is actually _not_ a valid topology on day 1. Assisted is not capable of creating an SNO with worker nodes.
But 3+1 is actually a valid topology, the bug report notwithstanding.